include/aspect/introspection.h

```diff
@@ -294,6 +294,28 @@ namespace aspect
       IndexSets index_sets;
 
       /**
+       * A struct that describes the available methods for
+       * compositional fields. This type is used to determine how
+       * to advect a compositional field.
+       */
+      struct FieldMethod
```

Comment:
Yes, I struggled with naming as well. I don't think I have a better suggestion -- `CompositionalFieldMethod` or `AdvectedFieldMethod` are both ok with me, depending on whether you want to include the temperature or not. I guess including the temperature makes sense if you want to use the thing here to discriminate between CG and DG methods.

I think all I wanted to say is that "FieldMethod" is just a bit too unspecific ;-)